staging/src/k8s.io/apimachinery/pkg/util/wait/wait.go

```diff
@@ -729,27 +734,30 @@ func poller(interval, timeout time.Duration) WaitWithContextFunc {
 
 // ExponentialBackoffWithContext works with a request context and a Backoff. It ensures that the retry wait never
 // exceeds the deadline specified by the request context.
-func ExponentialBackoffWithContext(ctx context.Context, backoff Backoff, condition ConditionFunc) error {
+func ExponentialBackoffWithContext(ctx context.Context, backoff Backoff, condition ConditionWithContextFunc) error {
```

Comment:
I duplicated this change in #115113, and the timer changes in poller I tackled in a slightly different way - I unified the underlying poll implementation and reused the timers, but the structure is different enough that it's a larger change.

Note that it is not required to drain the channel of a timer after a Stop() unless you plan to reuse the timer - this is a subtle aspect of Timer that also confused me.  So in #107826 I moved the timer to a top level defer in the loop for clarity.